Tests/ComposableArchitectureTests/EffectThrottleTests.swift

```diff
@@ -104,13 +112,96 @@ final class EffectThrottleTests: XCTestCase {
 
     runThrottledEffect(value: 5)
 
-    // A third value is throttled.
-    XCTAssertEqual(values, [1])
+    scheduler.advance(by: 0.25)
+
+    // The second (throttled) value emits.
+    XCTAssertEqual(values, [1, 2])
 
     scheduler.advance(by: 0.25)
 
-    // The first throttled value emits.
+    runThrottledEffect(value: 6)
+
+    scheduler.advance(by: 0.50)
+
+    // A third value is throttled.
     XCTAssertEqual(values, [1, 2])
+
+    runThrottledEffect(value: 7)
+
+    scheduler.advance(by: 0.25)
+
+    // The third (throttled) value emits.
+    XCTAssertEqual(values, [1, 2, 6])
```

Comment:
This would previously fail as values would contain `[1, 2, 2]` (the previous interval's first wasn't cleared)